core/src/commonMain/kotlin/com/powersync/utils/Json.kt

```diff
@@ -12,3 +15,18 @@ internal object JsonUtil {
     }
 }
 
+internal fun convertMapToJson(map: Map<String, Any>?): JsonObject {
+    if (map == null) return JsonObject(emptyMap())
+
+    val result = map.mapValues { (_, value) ->
+        when (value) {
+            is Int -> JsonPrimitive(value)
+            is Long -> JsonPrimitive(value)
+            is Double -> JsonPrimitive(value)
+            is Boolean -> JsonPrimitive(value)
+            is String -> JsonPrimitive(value)
```

Comment:
Are arrays/lists supported?

Reply:
Yeah you're right I've added support for Arrays, Lists and Maps now as well